tests/jballerina-unit-test/src/test/resources/test-src/typedefs/type-definitions-semantics-negative.bal

```diff
@@ -36,3 +36,9 @@ function bar() returns MyTuple => [1, ""];
 function testTypeReference() {
     function () returns int x = bar;
 }
+
+type Foo boolean|null;
+
+function testNullFiniteType() {
+    Foo _ = "null"; // error
```

Comment:
Shall we include a test case to import this type from a different module, just to test the BIR paths also?